packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -13,6 +15,25 @@ export type Data = Record<string, unknown>
 export type PaymentData = Data
 export type PaymentSessionData = Data
 
+export type PaymentContext = {
+  cart: {
+    context: Record<string, unknown>
+    id: string
+    email: string
+    shipping_address: Address | null
+    shipping_methods: ShippingMethod[]
+  }
+  currency_code: string
+  amount: number
+  resource_id?: string
+  collected_data: Record<string, unknown>
+}
+
+export type PaymentSessionResponse = {
+  collected_data: { customer: Record<string, unknown> }
+  session_data: Record<string, unknown>
+}
+
```

Comment:
thought: I think `collected_data` is an odd name. The PaymentProvider responds with some data that it wants the core to store. I think `update_requests` would be more suitable to explain the interaction between the core and the plugin. So:

```ts
export type PaymentSessionResponse = {
  update_requests: { customer: Record<string, unknown> } // maybe even { customer_metadata: Record<string, unknown> } to make it very clear where the data will be stored
  session_data: Record<string, unknown>
}
```

Furthermore, I think we should revise the PaymentContext to be:

```ts
export type PaymentContext = {
  cart: {
    context: Record<string, unknown>
    id: string
    email: string
    shipping_address: Address | null
    billing_address: Address | null // added
    shipping_methods: ShippingMethod[]
  }
  currency_code: string
  amount: number
  resource_id?: string
  customer?: Customer // changed - we don't yet have a notion of collected data, and it is reasonable always to provide the customer if we know it
}
```

--

I think we have confused two separate ideas. Let me know if this makes sense to you?

Reply:
^ +1

Reply:
I am alright to change the naming, i though if passing the collected data because the plugin consumer is not suppose to know where they are store. But know that I think about it, the doc will be able to say that they will always be in the metadata 😅 otherwise agree with you.

Reply:
included [here](https://github.com/medusajs/medusa/pull/2743/commits/5b30e2016c3ff51e2dc482d78f2f9f975a115326)